rllib/algorithms/algorithm_config.py

```diff
@@ -3343,402 +3352,199 @@ def get_evaluation_config_object(
+                * self.rollout_fragment_length
+            )
+            batch_size = min_batch_size
+            while batch_size < self.total_train_batch_size:
+                batch_size += min_batch_size
+            if batch_size - self.total_train_batch_size > (
+                0.1 * self.total_train_batch_size
+            ) or batch_size - min_batch_size - self.total_train_batch_size > (
+                0.1 * self.total_train_batch_size
+            ):
+                suggested_rollout_fragment_length = self.total_train_batch_size // (
+                    self.num_envs_per_env_runner * (self.num_env_runners or 1)
+                )
+                raise ValueError(
+                    "Your desired `total_train_batch_size` "
+                    f"({self.total_train_batch_size}={self.num_learners} "
+                    f"learners x {self.train_batch_size_per_learner}) "
+                    "or a value 10% off of that cannot be achieved with your other "
+                    f"settings (num_env_runners={self.num_env_runners}; "
+                    f"num_envs_per_env_runner={self.num_envs_per_env_runner}; "
+                    f"rollout_fragment_length={self.rollout_fragment_length})! "
+                    "Try setting `rollout_fragment_length` to 'auto' OR to a value of "
+                    f"{suggested_rollout_fragment_length}."
+                )
+
+    def get_torch_compile_worker_config(self):
+        """Returns the TorchCompileConfig to use on workers."""
+
+        from ray.rllib.core.rl_module.torch.torch_compile_config import (
+            TorchCompileConfig,
+        )
+
+        return TorchCompileConfig(
+            torch_dynamo_backend=self.torch_compile_worker_dynamo_backend,
+            torch_dynamo_mode=self.torch_compile_worker_dynamo_mode,
+        )
+
+    def get_default_rl_module_spec(self) -> RLModuleSpecType:
+        """Returns the RLModule spec to use for this algorithm.
+
+        Override this method in the subclass to return the RLModule spec, given
+        the input framework.
+
+        Returns:
+            The RLModuleSpec (or MultiRLModuleSpec) to
+            use for this algorithm's RLModule.
+        """
+        raise NotImplementedError
+
+    def get_default_learner_class(self) -> Union[Type["Learner"], str]:
+        """Returns the Learner class to use for this algorithm.
+
+        Override this method in the sub-class to return the Learner class type given
+        the input framework.
+
+        Returns:
+            The Learner class to use for this algorithm either as a class type or as
+            a string (e.g. ray.rllib.core.learner.testing.torch.BC).
+        """
+        raise NotImplementedError
+
+    def get_rl_module_spec(
+        self, env=None, spaces=None, inference_only=None
+    ) -> RLModuleSpec:
+        rl_module_spec = copy.deepcopy(self.rl_module_spec)
+
+        # If a MultiRLModuleSpec -> Reduce to single-agent (and assert that
```

Comment:
In which cases do we have a module IDs only on the learner? Just curious. Exploration modules?

Reply:
Correct, in cases where you do have multiple IDs as per the user's config (e.g. a default policy for learning how to act for some single-agent AND e.g. another module only used for learning, like a curiosity world-model that computes intrinsic rewards).